sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -73,6 +73,16 @@ trait BroadcastExchangeLike extends Exchange {
 
   protected def completionFuture: scala.concurrent.Future[broadcast.Broadcast[Any]]
 
+  /**
+   * Cancels broadcast job with a reason.
+   */
+  final def cancelBroadcastJob(reason: String): Unit = {
```

Comment:
Likewise, any way that we can factor out the common duplicated code here as well?